MapleServer2/Servers/Game/GameSession.cs

```diff
@@ -47,10 +44,10 @@ public void InitPlayer(Player player)
         public void EnterField(Player player)
         {
             // If moving maps, need to get the FieldManager for new map
-            if (player.MapId != FieldManager.MapId)
+            if (player.MapId != FieldManager.MapId || player.InstanceId != FieldManager.InstanceId)
             {
                 FieldManager.RemovePlayer(this, FieldPlayer); // Leave previous field
-                FieldManagerFactory.Release(FieldManager.MapId, player.InstanceId);
+                FieldManagerFactory.Release(FieldManager.MapId, FieldManager.InstanceId);
```

Comment:
```suggestion
                if (FieldManager.Release() <= 0) {
                    FieldManagerFactory.Release(FieldManager.MapId, FieldManager.InstanceId);
                }
```

Goes with the other comment.

Reply:
Also, if you make this change, then you might need to lock over this entire block.